mifosng-android/src/main/java/com/mifos/mifosxdroid/adapters/IndividualCollectionSheetAdapter.java

```diff
@@ -0,0 +1,282 @@
+        int position;
+        BulkRepaymentTransactions transaction = new BulkRepaymentTransactions();
+
+        public ViewHolder(View v) {
+            super(v);
+            ButterKnife.bind(this, v);
+            btnAdditional.setOnClickListener(this);
+        }
+
+        @Override
+        public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+
+            if (i != paymentTypeList.size() - 1) {
+                transaction.setPaymentTypeId(paymentTypeOptionsList.get(i).getId());
+            } else {
+                transaction.setPaymentTypeId(null);
+            }
+        }
+
+        @Override
+        public void onNothingSelected(AdapterView<?> adapterView) {
+
+        }
+
+        @Override
+        public void onClick(View view) {
+            position = getAdapterPosition();
+            switch (view.getId()) {
+                case R.id.btn_additional_details:
+                    if (tableAdditional.getVisibility() == View.VISIBLE) {
+                        tableAdditional.setVisibility(View.GONE);
+                    } else if (tableAdditional.getVisibility() == View.GONE) {
+                        showAdditional();
+                    }
+                    break;
+
+                case R.id.btn_cancel_additional:
+                    cancelAdditional();
+                    break;
+
+                case R.id.btn_save_additional:
+                    saveAdditional();
+                    break;
+            }
+        }
+
+        void saveAdditional() {
+
+            transaction.setLoanId(loans.get(position).getLoan().getLoanId());
+            transaction.setTransactionAmount((!etCharges.getText().toString().isEmpty() ?
+                    Double.parseDouble(etCharges.getText().toString()) : 0)
+                    + (!etTotalDues.getText().toString().isEmpty() ?
+                    Double.parseDouble(etTotalDues.getText().toString()) : 0));
+
+            if (!etAccountNumber.getText().toString().isEmpty()) {
+                transaction.setAccountNumber(etAccountNumber.getText().toString());
+            }
+
+            if (!etChequeNumber.getText().toString().isEmpty()) {
+                transaction.setCheckNumber(etChequeNumber.getText().toString());
+            }
+
+            if (!etRoutingCode.getText().toString().isEmpty()) {
+                transaction.setRoutingCode(etRoutingCode.getText().toString());
+            }
+
+            if (!etReceiptNumber.getText().toString().isEmpty()) {
+                transaction.setReceiptNumber(etReceiptNumber.getText().toString());
+            }
+
+            if (!etBankNumber.getText().toString().isEmpty()) {
+                transaction.setBankNumber(etBankNumber.getText().toString());
+            }
+
+            sheetItemClickListener.onSaveAdditionalItem(transaction, position);
+            tableAdditional.setVisibility(View.GONE);
+        }
+
+        void cancelAdditional() {
```

Comment:
These methods are local to this class, so make them private, look below methods too.